pkgs/by-name/ze/zepp-simulator/package.nix

```diff
@@ -0,0 +1,189 @@
+{
+  stdenv,
+  lib,
+  fetchurl,
+  makeWrapper,
+  makeDesktopItem,
+  copyDesktopItems,
+  autoPatchelfHook,
+
+  # Upstream is officialy built with Electron 18
+  # (but it works with Electron 24-30 with minor changes, see HACK below)
+  electron_30,
+  asar,
+  dpkg,
+
+  # qemu deps
+  glib,
+  libgcc,
+  libcxx,
+  zlib,
+  libepoxy,
+  libpng,
+  libaio,
+  xorg,
+  libvterm,
+  vte,
+  gsasl,
+  gtk3,
+  cairo,
+  gdk-pixbuf,
+  numactl,
+  cyrus_sasl,
+  SDL2,
+}:
+
+let
+  # CDN links for 2.0.2:
+  # MacOS: https://upload-cdn.zepp.com/zepp-applet-and-wechat-applet/20240927/ecb6ca54f4dc97a2f91e53358bbb532d.dmg
+  # Windows: https://upload-cdn.zepp.com/zepp-applet-and-wechat-applet/20240927/9db7ae1c60c26836a447a71a6fb25b3b.exe
+  # Linux ARM64: https://upload-cdn.zepp.com/zepp-applet-and-wechat-applet/20240927/02ec69e6a2f3b744d964fd7ba4f40fc3.deb
+  # Linux AMD64: https://upload-cdn.zepp.com/zepp-applet-and-wechat-applet/20240927/3e688d423cd0cd31a8a589b8325a309e.deb
+  srcs = {
+    x86_64-linux = {
+      url = "https://upload-cdn.zepp.com/zepp-applet-and-wechat-applet/20240927/3e688d423cd0cd31a8a589b8325a309e.deb";
+      sha256 = "sha256-ZHqaEL8FoSnRtuqGWpTyJka7D0dHtRADZthq8DG2k24=";
+    };
+    aarch64-linux = {
+      url = "https://upload-cdn.zepp.com/zepp-applet-and-wechat-applet/20240927/02ec69e6a2f3b744d964fd7ba4f40fc3.deb";
+      sha256 = "sha256-J5Y4wLiFOM9D2MIMiRyUtHIZ19rt65ktVCOMZQQwBCI=";
+    };
+  };
+
+in
+
+stdenv.mkDerivation {
+  pname = "zepp-simulator";
+  version = "2.0.2";
+
+  src = fetchurl srcs.${stdenv.hostPlatform.system};
+
+  patches = [
+    # Fix for qemu input grab not working with NIXOS_OZONE_WL=1
+    ./0001-force_qemu_x11.patch
+  ];
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+    makeWrapper
+
+    dpkg
+    asar
+
+    # QEMU deps (runtime):
+    glib
+    libgcc
+    libcxx
+    zlib
+    libepoxy
+    libpng
+    libaio
+    xorg.libX11
+    libvterm
+    vte
+    gsasl
+    numactl
+    cyrus_sasl
+    gtk3
+    cairo
+    gdk-pixbuf
+    SDL2
+  ];
+
+  dontBuild = true;
+
+  unpackPhase = ''
+    runHook preUnpack
+
+    dpkg-deb -x $src .
+
+    runHook postUnpack
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    # Create output file strucure
+    mkdir -p $out/{bin,opt,share}
+    cp -r opt/simulator $out/opt
+    cp -r usr/share/icons $out/share
+
+    # Remove unnecessary files
+    rm -rf \
+      $out/usr/share/applications/simulator.desktop \
```

Comment:
probably needs a patch tho, iirc it has path hardcoded to /opt

Reply:
You could use the following snippet in postPatch:

```
substituteInPlace --replace-fail '/opt/simulator/simulator' 'simulator'
```

Note that we want to avoid absolute paths in desktop files: #308324

Reply:
Also the upstream desktop file has a couple of other issues, the `MimeTypes` contains duplicate entries and doesn't specify the category properly